gensim/models/keyedvectors.py

```diff
@@ -147,9 +147,10 @@ def save_word2vec_format(self, fname, fvocab=None, binary=False, total_vec=None)
             for word, vocab in sorted(iteritems(self.vocab), key=lambda item: -item[1].count):
                 row = self.syn0[vocab.index]
                 if binary:
+                    row = row.astype(REAL)
```

Comment:
why this needed?

Reply:
Because 
```
from gensim.models.keyedvectors import KeyedVectors
model = KeyedVectors.load_word2vec_format('./test_data/test.kv.txt', datatype=np.float16)
print(model['horse.n.01'][0])
model.save_word2vec_format('./test_data/test.kv.bin', binary=True)
model2 = KeyedVectors.load_word2vec_format('./test_data/test.kv.bin', datatype=np.float32, binary=True)
```
this causes crash.
This is another bug that exists in develop branch.
Steps to reproduce error:
1. load model with low precision.
2. save that model in binary
3. then finally try to load the binary model.

Reply:
Ah, thanks for investigating that and reporting it. Do you have any ideas why that might be occurring?